worker/src/main/java/gov/cms/ab2d/worker/processor/domainmodel/ContractEobManager.java

```diff
@@ -0,0 +1,233 @@
+        String errMsg = ExceptionUtils.getRootCauseMessage(e);
+        OperationOutcome operationOutcome = FHIRUtil.getErrorOutcome(errMsg);
+        IParser jsonParser = fhirContext.newJsonParser();
+        String payload = jsonParser.encodeResourceToString(operationOutcome) + System.lineSeparator();
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        byteArrayOutputStream.write(payload.getBytes(StandardCharsets.UTF_8));
+        helper.addError(data);
+    }
+
+    /**
+     * returns true if the patient is a valid member of a contract, false otherwise. If either value is empty,
+     * it returns false
+     *
+     * @param benefit - The benefit to check
+     * @param patient - the patient who was used to create the query
+     * @return true if this patient is a member of the correct contract
+     */
+    boolean validPatientInContract(ExplanationOfBenefit benefit, ContractBeneficiaries.PatientDTO patient) {
+        if (benefit == null || patient == null) {
+            log.debug("Passed an invalid benefit or an invalid list of patients");
+            return false;
+        }
+        String patientId = benefit.getPatient().getReference();
+        if (patientId == null) {
+            return false;
+        }
+        patientId = patientId.replaceFirst("Patient/", "");
+        return patientId.equalsIgnoreCase(patient.getPatientId());
+    }
+
+    Map<String, EobSearchResponse> getUnknownEobs() {
+        return unknownEobs;
+    }
+
+    Map<String, EobSearchResponse> getValidEobs() {
+        return validEobs;
+    }
+
+    public void cleanUpKnownInvalidPatients(List<Integer> monthsDone) {
+        for (Map.Entry<String, EobSearchResponse> entry: unknownEobs.entrySet()) {
+            EobSearchResponse response = entry.getValue();
+            List<Resource> resources = response.getResources();
+            response.setResources(resources.parallelStream()
+                    .filter(r -> !covered(r, monthsDone))
+                    .collect(Collectors.toList()));
+        }
+    }
+
+    static boolean covered(Resource resource, List<Integer> monthsDone) {
+        ExplanationOfBenefit eob = (ExplanationOfBenefit) resource;
+        List<Integer> coveredMonths = getCoveredMonths(eob.getBillablePeriod());
+        boolean done = false;
+        for (int i = 0; i < coveredMonths.size(); i++) {
+            if (monthsDone.contains(coveredMonths.get(i))) {
+                done = true;
+            }
+        }
+        return done;
+    }
+
+    static List<Integer> getCoveredMonths(Period period) {
+        Date startDate = period.getStart();
+        int startMonth = getMonth(startDate);
+        int startYear = getYear(startDate);
+        Date endDate = period.getEnd();
+        int endMonth = getMonth(endDate);
+        int endYear = getYear(endDate);
+        List<Integer> allMonths = List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12);
```

Comment:
Small change, but seems like allMonths can move to the class level with it being read only to avoid initializing every time.